_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,493 @@
+Client can be used for instance:
+
+```
+// 1. Creating the client instance 
+
+BasicExternalKafkaClient basicExternalKafkaClient = new BasicExternalKafkaClient.Builder()
+    .withTopicName("topic-name")
+    .withNamespaceName("namespace-name")
+    .withClusterName("cluster-name")
+    .withKafkaUsername("kafka-username")
+    .withMessageCount(100)
+    .withSecurityProtocol(SecurityProtocol.SSL)
+    .build();
+
+// 2. Sending messages
+
+basicExternalKafkaClient.sendMesssageTls();
+```
+
+## <a id="dependencies">Dependencies</a>
+
+Before writing simple test case in Strimzi system tests, we need to know main dependencies that `system test` module has. 
+The most important ones are:
+1. test (kubernetes client)
+2. crd-generator + api (chaining / builder pattern)
+
+### test module
+
+In the test module, we have the Kubernetes client, which we use in the system tests to create, list, delete and update 
+Kubernetes resources such as Service, Deployment, Pod, ConfigMap and so on. 
+Basically, we encapsulate the [Kubernetes client](https://github.com/fabric8io/kubernetes-client) from fabric8 to use 
+client more handy with not additional and typically redundant method chaining. 
+Moreover, we have our abstraction of Kubernetes cluster and mechanism to detect on which type is currently client. 
+Clusters could be Kubernetes, Minikube and so on.
+Lastly, in this module you can find [Command Client](https://github.com/strimzi/strimzi-kafka-operator/tree/master/test/src/main/java/io/strimzi/test/k8s/cmdClient),
+which servers as executor and can invoke various requests such as `kubectl annotate...`, `kubectl exec...` and so on.
+
+### crd-generator + api module
+
+Resources previously described in Resource section are used in crd-generator and api modules.
+Every time, when the code of Model changes you have to build the whole modules to generate new schema for resources. 
+For instance, imagine that you are using `Kafka` resource and you need to create Kafka with the `Cruise Control` support but you have old schemas, which does not support it.
+You will need to build these modules and generate schema, which will support it and be able to create Kafka with Cruise Control. 
+These modules provide the fluent way of creating Strimzi custom resource instances such as `Kafka`, `KafkaConnect`, `KafkaConnectS2I`, `KafkaMirrorMaker` and many other.
+
+## <a id="createtest">How to create a system test</a>
+
+At this point you have the necessary knowledge to create a simple test. There are also some additional steps, which need to be performed to set up your testing environment.
+
+### Precursors
+
+1. Fork the `strimzi-kafka-operator` repository - https://github.com/strimzi/strimzi-kafka-operator
+2. Build the whole project in root directory - `mvn clean install -DskipTests` or you can use `mvn -am clean install -DskipTests -f systemtest/pom.xml`
+3. Setup your Kubernetes cluster
+4. Login to your Kubernetes cluster
```

Comment:
Should you mentioned the priviledges needed to install Strimzi (not list all of them, but say _Login to your Kubernetes cluster as a user who has the rights to install the Strimzi operator_)